lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -474,7 +476,7 @@ def _normalize_galaxy_yml_manifest(
 
     set_keys = set(galaxy_yml.keys())
     missing_keys = mandatory_keys.difference(set_keys)
-    if missing_keys:
+    if missing_keys and require_mandatory_keys:
         raise AnsibleError("The collection galaxy.yml at '%s' is missing the following mandatory keys: %s"
```

Comment:
What if instead of ignoring the problems we'd accumulate them in a list and implement a postponed console output + erroring out?

Reply:
That seems like a good idea. I'll take a stab at that tomorrow.

Reply:
you can output warnings immediately (they dedupe) and that should not interrupt the other output that happens later

Reply:
Yeah, I think just warning is the best solution. Trying to shuffle around the errors to accumulate them at the end got pretty messy. Now there is a warning for galaxy.yml's that aren't a MutableMappings and for missing keys required for building.

My patch looks strange because it's displaying a warning and then raising a ValueError with the same error, but we silently ignore the ValueErrors (in `_ComputedReqKindsMixin.from_dir_path_as_unknown`), and use them to fall back to the path-derived metadata. Adding warnings for ValueErrors in `_ComputedReqKindsMixin.from_dir_path_as_unknown` would make sense to me but then `_ComputedReqKindsMixin.from_dir_path` and maybe other things need to be refactored to not display extra warnings. So this is maybe something to refactor in the future.